tests/decorators/test_external_python.py

```diff
@@ -125,3 +125,49 @@ def f(_):
             ret = f(datetime.datetime.utcnow())
 
         ret.operator.run(start_date=DEFAULT_DATE, end_date=DEFAULT_DATE)
+
+    def test_marking_external_python_task_as_setup(self, dag_maker, venv_python):
+        @setup
+        @task.external_python(python=venv_python)
+        def f(_):
+            return 1
+
+        with dag_maker() as dag:
+            ret = f(datetime.datetime.utcnow())
```

Comment:
```suggestion
        def f():
            return 1

        with dag_maker() as dag:
            ret = f()
```

Not sure we need this arg? Same with the rest of these.